merlion/models/automl/autoets.py

```diff
@@ -30,3 +92,166 @@ class AutoETS(SeasonalityLayer):
     """
 
     config_class = AutoETSConfig
+
+    def generate_theta(self, train_data: TimeSeries) -> Iterator:
+        """
+        generate [theta]. theta is a list of parameter combination [error, trend, damped_trend, seasonal]
+        """
+        y = train_data.univariates[self.target_name].np_values
+
+        # check the size of y
+        n_samples = y.shape[0]
+        if n_samples <= 3:
+            self.information_criterion = "aic"
+
+        # auto-detect seasonality if desired, otherwise just get it from seasonal order
+        if self.config.auto_seasonality:
+            candidate_m = super().generate_theta(train_data=train_data)
+            self.config.seasonal_periods, _, _ = super().evaluate_theta(thetas=candidate_m, train_data=train_data)
+
+
+        # set the parameters ranges for error, trend, damped_trend and seasonal
+        if np.any(y <= 0):
+            E_range = ["add"]
+            T_range = ["add", None]
+        else:
+            E_range = ["add", "mul"]
+            if self.config.allow_multiplicative_trend:
+                T_range = ["add", "mul", None]
+            else:
+                T_range = ["add", None]
+
+        if self.config.seasonal_periods <= 1 or self.config.seasonal_periods is None or y.shape[0] <= self.config.seasonal_periods:
+            self.config.seasonal_periods = 1
```

Comment:
This causes errors in the test cases. `seasonal_periods` needs to be `None`, not 1 in this case. You should also first check if `seasonal_periods` is `None` before checking if `seasonal_periods <= 1` to avoid potential type errors.